prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,103 @@
+# Prometheus Alerting Consistency
+
+## Summary
+
+This guide is originally based on the [Alerting Consistency][1] proposal for OpenShift alerts.
+
+Clear and actionable alerts are a key component of a smooth operational experience.
+Ensuring you have clear and concise guidelines for engineers and SRE creating new alerts will result in a better experience for end users.
+
+## Recommended Reading
+
+A list of references on good alerting practices:
+
+* [Google SRE Book - Monitoring Distributed Systems][2]
+* [Prometheus Alerting Documentation][3]
+* [Alerting for Distributed Systems][4]
+
+## Style Guide
+
```

Comment:
Maybe (a very loose maybe :) a worthy addition as a tip: we have also found it useful for us to name the alert with the same name as the SOP. That is as a 1-1 relationship. This helps us find the SOP if the there is some "problem" with the sop_url field. Or in the case where we see the alert firing in alertmanager but it has not yet sent its notification.

Reply:
@brizrobbo Do you have an example?
Which came first in this case, the SOP or the Alert? (Was the SOP named after the alert?)
The reason I'm asking, if I understand what you're saying correctly, is because it would introduce a different naming convention that highlights the 'fix' rather than the 'problem'

Reply:
For examples, all the filenames for the RHOAM alerts follow this convention. 
In all cases the alert was named first, and then the SOP simply mirrored that.
There is one complication: what if n-alerts followed the same SOP? For us, that is OK, as we create a SOP per alert anyway even if it is the same .... this makes it easy to differentiate it later if we need to....however, it does create an extra maintenance overhead.

Reply:
@brizrobbo Is this something you want to include in the consistency guide and push for other teams to use?

Reply:
Recommend we leave this detail out for now.  This feels like something we'd implement if alerts, SOPs etc were auto-generated by some tooling.